.github/workflows/monodevelop.yml

```diff
@@ -4,20 +4,138 @@ on:
+    container:
+      image: "centos:centos8"
     steps:
+    - name: Dependencies
+      run: |
+        dnf update
+        dnf install -y sudo
+        sudo dnf install -y dotnet-sdk-3.1
+        rpmkeys --import "http://pool.sks-keyservers.net/pks/lookup?op=get&search=0x3fa7e0328081bff6a14da29aa6a19b38d3d831ef"
+        su -c 'curl https://download.mono-project.com/repo/centos8-stable.repo | tee /etc/yum.repos.d/mono-centos8-stable.repo'
+        sudo dnf install -y gtk-sharp2 glade-sharp2 monodoc-core git
+        sudo dnf install -y fsharp --nobest
+        sudo dnf group install -y "Development Tools"
+        git --version
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        submodules: true
+    - name: Configure and build
+      run: |
+        ./configure --profile=gnome
+        make
+
+  ubuntu2004:  
+    runs-on: ubuntu-latest
+    container:
+      image: "ubuntu:20.04"
+    steps:
+    - name: Dependencies
+      run: |
+        apt-get update
+        apt-get install -y sudo        
+        sudo apt-get update
+        sudo apt-get install -y wget
+        wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+        sudo dpkg -i packages-microsoft-prod.deb
+        sudo apt-get install -y apt-transport-https
+        sudo apt-get update
+        sudo apt-get install -y dotnet-sdk-3.1
+        sudo apt-get install -y gnupg ca-certificates
+        sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
+        echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list
+        sudo apt-get update
+        sudo apt-get install -y sed build-essential intltool fsharp gtk-sharp2
+        sudo apt-get install -y software-properties-common
+        sudo add-apt-repository ppa:git-core/ppa -y
+        sudo apt-get update
+        sudo apt-get install -y git
+        git --version
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        submodules: true
+    - name: Configure and build
+      run: |
+        ./configure --profile=gnome
+        make    
+
+  ubuntu1804:  
+    runs-on: ubuntu-latest
+    container:
+      image: ubuntu:18.04
+    steps:
+    - name: Dependencies
+      run: |
+        apt-get update
+        apt-get install -y sudo        
+        sudo apt-get update
+        sudo apt-get install -y wget
+        wget https://packages.microsoft.com/config/ubuntu/18.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+        sudo dpkg -i packages-microsoft-prod.deb
+        sudo apt-get install -y apt-transport-https
+        sudo apt-get update
+        sudo apt-get install -y dotnet-sdk-3.1
+        sudo apt-get install -y gnupg ca-certificates
+        sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
```

Comment:
@lextm this script fragment is triplicated, should we create a sh script to avoid some duplication?

Reply:
@knocte that's a good idea. Let's use this pull request to collect feedback only. I plan to open a new pull request later with some updates.